src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -77,11 +81,16 @@ public CommandResult execute(Model model) throws CommandException {
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+        if (personToEdit.equals(editedPerson)) {
+            throw new CommandException(MESSAGE_NO_CHANGE);
+        }
+        model.deletePerson(personToEdit);
+        if (model.hasPerson(editedPerson)) {
+            model.addPersonForEdit(personToEdit);
             throw new CommandException(MESSAGE_DUPLICATE_PERSON);
         }
 
-        model.setPerson(personToEdit, editedPerson);
```

Comment:
For the regular addPerson, it will reset the filter to display all the contacts.
For a user who has filtered by a certain criteria, then edited a person in the list, it might be confusing if the list is suddenly updated, causing the person they were updating to disappear from view.

Therefore addPersonForEdit keeps the current filter. Though if the user has filtered by name, then the displayed list will change.
*Rename addPersonForEdit to AddPersonKeepFilter for clarity.

Reply:
Thank you for the clarification!